spec/system/admin/bulk_url_upload_spec.rb

```diff
@@ -1,130 +1,107 @@
-    allow_any_instance_of(SearchgovDomain).to receive(:index_urls) do |searchgov_domain|
-      @reindexed_domains << searchgov_domain
-    end
-  end
+  before { allow(searchgov_domain).to receive(:index_urls) }
 
   it_behaves_like 'a page restricted to super admins'
 
-  describe 'bulk uploading a file of URLs' do
+  context 'when a super admin is logged in' do
     include_context 'log in super admin'
 
-    it 'sends us back to the bulk upload page' do
-      bulk_upload
-      expect(page).to have_text('Bulk Search.gov URL Upload')
-    end
-
-    it 'shows a confirmation message' do
-      bulk_upload
-      expect(page).to have_text(
-        <<~CONFIRMATION_MESSAGE
-          Successfully uploaded #{url_filename} for processing.
-          The results will be emailed to you.
-        CONFIRMATION_MESSAGE
-      )
-    end
+    describe 'bulk uploading a file of URLs' do
+      let(:upload_filename) { 'good_url_file.txt' }
 
-    it 'creates the URLs' do
-      bulk_upload
-      urls.each do |url|
-        expect(SearchgovUrl.find_by(url: url)).not_to be_blank
+      it 'sends us back to the bulk upload page' do
+        do_bulk_upload
+        expect(page).to have_text('Bulk Search.gov URL Upload')
       end
-    end
-
-    it 're-indexes the domains for the URLs' do
-      bulk_upload
-      expect(@reindexed_domains).to eq(searchgov_domains)
-    end
-  end
 
-  describe 'trying to bulk upload a file of URLs when there is no file attached' do
-    include_context 'log in super admin'
+      it 'shows a confirmation message' do
+        do_bulk_upload
+        expect(page).to have_text(
+          <<~CONFIRMATION_MESSAGE
+            Successfully uploaded #{upload_filename} for processing.
+            The results will be emailed to you.
+          CONFIRMATION_MESSAGE
+        )
+      end
 
-    subject(:bulk_upload) do
-      visit url
-      click_button('Upload')
-    end
+      it 'creates the URLs' do
+        do_bulk_upload
 
-    it 'sends us back to the bulk upload page' do
-      bulk_upload
-      expect(page).to have_text('Bulk Search.gov URL Upload')
-    end
+        urls.each do |url|
+          expect(SearchgovUrl.find_by(url: url)).not_to be_blank
+        end
+      end
 
-    it 'shows an error message' do
-      bulk_upload
-      expect(page).to have_text(
-        <<~ERROR_MESSAGE
-          Please choose a file to upload.
-        ERROR_MESSAGE
-      )
+      it 're-indexes the domains for the URLs' do
+        reindexed_domains = Set.new
+        allow_any_instance_of(SearchgovDomain).to receive(:index_urls) do |searchgov_domain|
+          reindexed_domains << searchgov_domain
+        end
+        do_bulk_upload
+        expect(reindexed_domains).to eq(Set[searchgov_domain])
+      end
     end
-  end
 
-  describe 'trying to bulk upload a file of URLs that is not a text file' do
-    include_context 'log in super admin'
+    context 'when bulk uploading a UTF-8 file of URLs' do
```

Comment:
see original comment - the description still needs to be changed
https://github.com/GSA/search-gov/pull/752#discussion_r678406103